app/controllers/recycle_locations_controller.rb

```diff
@@ -2,20 +2,28 @@ class RecycleLocationsController < ApplicationController
   before_action :authenticate_with_token, only: [:update]
 
   def index
-    if params[:latitude].present? && params[:longitude].present?
-      @recycle_locations =
-        RecycleLocation.near([latitude, longitude], 40, units: :km).limit(75)
-    else
-      head :bad_request
-    end
+    CoordinateValidation.validate(params)
+
+    @recycle_locations =
+      RecycleLocation.near([latitude, longitude], 40, units: :km).limit(75)
+    respond_with(@recycle_locations)
+
+    rescue Exceptions::CoordinatesNotPresent => e
+      respond_with( error_message(e, "No coordinates provided in params") )
```

Comment:
It's better if you `raise` with a message and then pass a long the message. For example:

    raise CoordinatesNotPresentError, "No coordinates provided"

Then you can rescue the error and grab the message immediately:

    rescue CoordinatesNotPresentError => message
      respond_with({ error: true, message: message })